packages/next/src/shared/lib/turbopack/compilation-events.ts

```diff
@@ -0,0 +1,41 @@
+import type { Project } from '../../../build/swc/types'
+import * as Log from '../../../build/output/log'
+
+/**
+ * Subscribes to compilation events for `project` and prints them using the
+ * `Log` library.
+ */
+export function backgroundLogCompilationEvents(
+  project: Project,
+  { eventTypes, signal }: { eventTypes?: string[]; signal?: AbortSignal } = {}
+) {
+  ;(async function () {
+    for await (const event of project.compilationEventsSubscribe(eventTypes)) {
+      if (signal?.aborted) {
```

Comment:
nit: we should really pass the abortsignal down, otherwise we only respect it after the next event arrives

Reply:
yeah add a comment.  it looks like at the napi layer this is somewhat implementable (we could stop the iterator), but fair that a true solution is more complex

a comment is a reasonable compromise